rosidl_generator_py/rosidl_generator_py/generate_py_impl.py

```diff
@@ -19,11 +19,30 @@
 from rosidl_cmake import expand_template
 from rosidl_cmake import get_newest_modification_time
 from rosidl_cmake import read_generator_arguments
-from rosidl_generator_c import primitive_msg_type_to_c
 from rosidl_parser import parse_message_file
 from rosidl_parser import parse_service_file
 
 
+def _primitive_msg_type_to_c(type_):
+    msg_type_to_c = {
+        'bool': 'bool',
+        'byte': 'uint8_t',
+        'char': 'uint8_t',
+        'float32': 'float',
+        'float64': 'double',
+        'uint8': 'uint8_t',
+        'int8': 'int8_t',
+        'uint16': 'uint16_t',
+        'int16': 'int16_t',
+        'uint32': 'uint32_t',
+        'int32': 'int32_t',
+        'uint64': 'uint64_t',
+        'int64': 'int64_t',
+        'string': 'rosidl_generator_c__String',
+    }
+    return msg_type_to_c[type_]
```

Comment:
@dirk-thomas I've seen this mapping replicated in multiple places, here and in https://github.com/ros2/rosidl/pull/334, but I'm sure I'm missing a part of the history.

Reply:
The mapping shouldn't be identical. They are in general specific to the generator.